src/main/java/org/pentaho/di/git/spoon/GitSpoonMenuController.java

```diff
@@ -69,13 +70,40 @@ public void openRepo() {
       if ( name == null ) {
         return;
       }
-      GitRepository repo = repoFactory.loadElement( name );
-      gitController.openGit( repo );
+
+      openRepo( repoFactory, name );
     } catch ( Exception e ) {
-      e.printStackTrace();
+      new ErrorDialog( Spoon.getInstance().getShell(), "Error", "Error opening Git Reposisory", e );
     }
   }
 
+  private void openRepo( MetaStoreFactory<GitRepository> repoFactory, String repositoryName ) throws MetaStoreException {
+    GitRepository repo = repoFactory.loadElement( repositoryName );
+    gitController.openGit( repo );
+  }
+
+  /**
+   * Open a repository with a given name.
+   * Convenience method so we can call this method from other places/plugins.
+   *
+   * @param repositoryName The name of the repository to open
+   * @throws MetaStoreException
+   */
+  public void openRepo( String repositoryName ) throws MetaStoreException {
+    MetaStoreFactory<GitRepository> repoFactory = getRepoFactory();
+    openRepo(repoFactory, repositoryName);
+  }
+
```

Comment:
Unfortunately github merged the pull requests.  No idea on how to prevent this.
I just need the convenience methods so I can call them from another plugin.  Anything else is up to you.  It would be great if we could have variable replacement for the git repository directory but for now I have my own build of this project so take your time.